File: Ilya-Dudnikov/term_2/block_4/task_2/bash/src/main/java/BashProject/shellcommand/commands/Pwd.java
package BashProject.shellcommand.commands;

import BashProject.shellcommand.Command;

import java.nio.ByteBuffer;

public class Pwd extends Command {
	@Override
	public ByteBuffer run(String ... args) {
		return ByteBuffer.wrap((System.getProperty("user.dir") + System.lineSeparator()).getBytes());
	}
}

File: ShangareevRustam/Second term/Block 4/2. Simple bash interpreter/app-bash/src/main/kotlin/bashCommands/Pwd.kt
package bashCommands

object Pwd : BashCommand() {

	override fun run(arg: String, input: String, arguments: MutableMap<String, String>): String {
		return System.getProperty("user.dir")
	}

	override fun isValid(command: String) =
		command.split(" ").any { it != "" } &&
				command.split(" ").first { it != "" } == "pwd"

}